### PR TITLE
Enable AppVeyor CI

### PR DESCRIPTION
**Pull request by vincent-psarga**
## Summary

Enable appveyor to run CI on Windows hosts too.

## Motivation and Context

Ensure tests are also executed on a Windows host to ensure compatibility.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I've added tests for my code
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by vincent-psarga** [comment]
@aslakhellesoy @tooky @mattwynne - could one of you check out this PR ?
I've had to update some of the tests so they run on Windows. From what I've understood of the code, the changes are correct but a second look would be great before merging that :)

**Comment by vincent-psarga** [comment]
I was thinking about doing a squash & merge (like I did for CircleCI).
For CI branches in general it works fine, the intermediary commits are not really meaningful.

File: spec/cucumber/core/test/location_spec.rb [hunk 1]
```diff
@@ -100,14 +100,17 @@ module Cucumber::Core::Test
...
+          expect( Location.from_source_location("/path/file.rb", 1).file ).to eq File.expand_path("/path/file.rb")
         end
       end
     end
 
     describe "created from file-colon-line" do
       it "handles also Windows paths" do
-        expect( Location.from_file_colon_line("c:\path\file.rb:123").file ).to eq "c:\path\file.rb"
+        # Note: running this test on Windows will produce "c:/path/file.rb", but "c:\path\file.rb" on Linux.
+        expect( Location.from_file_colon_line("c:\\path\\file.rb:123").file ).to match(/c:(\\|\/)path(\\|\/)file.rb/)
```
Comment (tooky) [inline spec/cucumber/core/test/location_spec.rb]: This looks reasonable - it makes me wonder if the Location.from_file_colon_line should return something more portable across platforms?

**Comment by aslakhellesoy** [comment]
Beautiful @vincent-psarga! Will we get slack notifications if the build starts failing again? If not, could you add that?

**Comment by vincent-psarga** [comment]
I'm not sure about Slack notifications, it seems like we don't have the app enabled on the Cucumber slack and we already have 10 apps installed.
Do they currently work ? I've never seen notifications coming from AppVeyor, but maybe it's on another channel than bots-monorepo ?

File: spec/cucumber/core/test/location_spec.rb (hunk 1 above)
Comment (luke-hill) [reply spec/cucumber/core/test/location_spec.rb]: This issue is something encountered in the selenium project (specifically many issues about this in Ruby), and the way in which they fix it varies in place to place and with some considerably hacky behaviour.

My personal opinion is we create some new kind of helper which returns an array with each dir level, as this covers each of the platforms.

Happy to listen to other POVs though

**Comment by luke-hill** [comment]
Quick query, after doing this can you look at https://github.com/cucumber/cucumber-ruby/pull/1351 / https://github.com/cucumber/cucumber-ruby/issues/1215 as these are causing a bit of a backlog when reviewing code for the ruby API

**Comment by vincent-psarga** [comment]
Yep sure, I have a look at that :)

**Comment by vincent-psarga** [comment]
I merge it as otherwise other PRs are marked as fail with CircleCI.

**Pull request merged**